### DISCUSSION TITLE
Working with multiple virtual environments

### DISCUSSION THREAD

**Discussion by NekiMedo** [reloadware/reloadium#8] [Q&A] [answered]
```
$ reloadium run ./myscript.py
■■■■■■■■■■■■■■■
Reloadium 0.8.6
■■■■■■■■■■■■■■■
If you like this project please support it by giving a star at https://github.com/reloadware/reloadium

Watching paths:
‣ /home/.../Code/<snip>/**/*.py

Traceback (most recent call last):
<snip>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
```

This otherwise works fine when launched from command line.

**Comment by dkrystki** [comment]
That's very strange. 
I could not reproduce it.
Could you add these line to your code (before you import numpy), run it and confirm that your venv path is in there?

```
import sys
print(sys.path)
```

**Reply by NekiMedo** [reply]
added code:
```
import sys
print( sys.path )
a = input( 'press ENTER' )
```
the output:
```
$ reloadium run ./udp_text_receiver.py 
■■■■■■■■■■■■■■■
Reloadium 0.8.6
■■■■■■■■■■■■■■■
If you like this project please support it by giving a star at https://github.com/reloadware/reloadium

Watching paths:
‣ /home/user/Code/MyProject/**/*.py

['/home/user/Code/MyProject', '/home/user/.local/bin', '/home/user/.local/bin', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/home/user/.local/lib/python3.10/site-packages', '/usr/local/lib/python3.10/dist-packages', '/usr/lib/python3/dist-packages']
press ENTERWatching 1 file.


Traceback (most recent call last):
  File "/home/user/Code/MyProject/udp_text_receiver.py", line 14, in <module>
    import my_tts as tts
  File "/home/user/Code/MyProject/vas_tts.py", line 8, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'

# without reloadium:
$ ./udp_text_receiver.py 
['/home/user/Code/MyProject', '/home/user/.local/bin', '/usr/lib/python39.zip', '/usr/lib/python3.9', '/usr/lib/python3.9/lib-dynload', '/home/user/virtualenv/TTS/lib/python3.9/site-packages']
press ENTER
```
likely it has to do with the version of Python being used; some modules i use don't work with Python 3.10 so i use Python 3.9.12 in virtual environment - which is what gets picked up when running without `reloadium`. Looks like `reloadium`  kicks off Python 3.10 - it should probably be picking the interpreter found at `$VIRTUAL_ENV/bin` PATH

BTW this is running on Debian bookworm (probably doesn't matter)

**Comment by dkrystki** [comment, marked as answer]
I think the issue here is that reloadium was installed in a wrong environment.
By invoking your script with `./udp_text_receiver.py ` your script is run by python3.9 (from the shebang) and by invoking with reloadium it uses venv where reloadium was installed which is 3.10.
I think if you install reloadium in 3.9 one it should work fine.

**Reply by NekiMedo** [reply]
you are right - removing the user-wide installation and re-installing it within the virtual environment sorted things out;
thanks for your help
